igcollect/rsyslog_impstats.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env python3
+"""igcollect - Rsyslog impstats parser
+
+Copyright (c) 2019 InnoGames GmbH
+"""
+
+import re
+from argparse import ArgumentParser
+from time import time
+
+def parse_log():
+    actions = {}
+    args = parse_args()
+
+    with open(args.filename, 'r') as pstats_file:
+        for line in pstats_file:
+            lastmatch = re.findall(r'(action-[0-9]+)(.*discarded\.full.*)', line)
+            if lastmatch:
+                actions [lastmatch[0][0]] = lastmatch[0][1]
+
+    # Extract the stats we need for the last matched line of each action
+    result = []
+
+    for action in actions: 
```

Comment:
maybe, `for action, base in actions.items()`?

Reply:
Should be fixed in new commit https://github.com/innogames/igcollect/pull/128/commits/f6e2541aba498c489c3c013379edf6b909546b03